src/g3log/logmessage.hpp

```diff
@@ -87,7 +88,8 @@ namespace g3 {
       // Complete access to the raw data in case the helper functions above
       // are not enough.
       //
-      std::time_t _timestamp;
+      // TODO: unique_ptr should be considered instead of shared_ptr
+      std::shared_ptr<Timestamp> _timestamp;
```

Comment:
What I meant was to have a function-ptr that returned a std::string formatted as desired.

The function-ptr should be called at the creation of the object. And the "time string" would be saved in the message object. I.e it would work with std::time as well as boost::...ptime 

However, I realize it would break the API. The logging sinks should be able to request the time string formatt d in any way possible
Example:
YYYY
YY:SS
YYYY:MM:DD HH:MM:SS

These three are all valid string formats of the time that a sink could use.

---